js/src/frameworks/openai.ts

```diff
@@ -132,18 +132,22 @@ export class OpenAIToolSet extends BaseComposioToolSet {
                 runId = id;
             }
 
-            if(!runId) {
+            if (!runId) {
                 continue;
             }
 
             // Handle the 'requires_action' event
             if (event.event === 'thread.run.requires_action') {
                 const toolOutputs = await this.handleAssistantMessage(event.data, entityId);
 
-                // Submit the tool outputs
-                await client.beta.threads.runs.submitToolOutputs(thread.id, runId, {
-                    tool_outputs: toolOutputs
-                });
+                // Check if the run is still in a state that accepts tool outputs
+                const currentRun = await client.beta.threads.runs.retrieve(thread.id, runId);
+                if (["queued", "in_progress", "requires_action"].includes(currentRun.status)) {
+                    // Submit the tool outputs
+                    await client.beta.threads.runs.submitToolOutputs(thread.id, runId, {
+                        tool_outputs: toolOutputs
+                    });
+                }
```

Comment:
Consider adding error logging when the run status is not in a valid state for tool output submission. This would help with debugging:
```typescript
} else {
    logger.warn(`Run ${runId} is in ${currentRun.status} state, skipping tool output submission`);
}
```